I added the feature importance function

zelioluca opened the pull request:
Hello Niko! How are you? Let's start with this that is simple :-)

zelioluca commented:
I passed the pre commit stuff already

nmaarnio commented:
Hi, thanks for the updates! I'll review it now.

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+    clf: sklearn.neural_network or sklearn.linear_model,
+    x_test: np.ndarray,
+    y_test: np.ndarray,
+    feature_names: list[str],
```
Is any sequence ok here? We have used `Sequence[str]` for many cases before to allow tuples etc.

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+    y_test: np.ndarray,
+    feature_names: list[str],
+    number_of_repetition: int = 50,
+    random_state: int = 0,
```
Should the default be 0 or None? I thought None would be good default as it is so in Sklearn, but just asking

zelioluca replied:
this is an int should be any number I always set it to 0

zelioluca commented:
Let s see let s see 😁

On Tue, 24 Oct 2023, 12:56 Niko Aarnio, ***@***.***> wrote:

> Hi, thanks for the updates! I'll review it now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GispoCoding/eis_toolkit/pull/211#issuecomment-1776894156>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF3KUFJZ7VTTRSWQXYJ2HU3YA6GFFAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZWHA4TIMJVGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+import numpy as np
+import pandas
```
We have used `import pandas as pd` so far (and I noticed there seems to be duplicate import). Sorry for nitcpicky comment, but if its ok you should use the same import alias as elsewhere

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+    Evaluate the feature importance of a sklearn classifier or linear model.
+
+    Parameters:
+        clf (Any sklearn nn model or lm model): Trained classifier.
```
Regarding the docstring, no need to include the type in paranthesis (just a decision we made).

zelioluca replied:
I removed type than!

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+        y_test (np.ndarray): Testing target data.
+        feature_names (list): Names of the feature columns.
+        number_of_repetition (int): Number of iteration used when calculate feature importance (default 50).
+        random_state (int): random state for repeatability of results (Default 0).
+    Return:
```
```suggestion
        random_state (int): random state for repeatability of results (Default 0).
    
    Return:
```

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+    Return:
+        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
+        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
+    Raise:
```
```suggestion
    
    Raises:
```

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+
+    feature_importance["Importance"] = feature_importance["Importance"] * 100
+    feature_importance = feature_importance.sort_values(by="Importance", ascending=False)
+    # feature_importance['Importance'] = feature_importance['Importance'].apply(lambda x: '{:.6f}%'.format(x))
```
Could remove this if its not needed for final implementation

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
```diff
@@ -0,0 +1,48 @@
+        number_of_repetition (int): Number of iteration used when calculate feature importance (default 50).
+        random_state (int): random state for repeatability of results (Default 0).
+    Return:
+        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
+        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
```
The names of the returned objects don't need to be stated, you can check some other docstrings for examples.

nmaarnio commented on tests/feature_importance/feature_importance.py:
The test file should use `pytest` and have some basic unit type tests. These are sometimes quite stupid to do for models, but you could load some test data or create some small test array and then see that the results are as you would expect. Again I suggest to take a look at some other test files.

Some features are nice to demonstrate as a notebook, but if this does not deserve its own one in the toolkit (up to you), we can combine some modeling related stuff in one notebook to verify they work and show people how to use them at the same time.

nmaarnio commented on .idea/eis_toolkit.iml:
Not sure what is the `.idea` folder, but you should probably remove it from the PR

nmaarnio commented on eis_toolkit/feature_importance/feature_importance.py:
Feature importance is marked in the excel at least to belong to Exploratory analyses, so you should move it there. Or if you think some other is more suitable thats ok too, but it shouldnt have its own folder directly under eis_toolkit folder

nmaarnio reviewed (changes requested):
Hi! The core functionality looks good to me and overall the style is almost according to the guideline, but I found a few small nitpicky style issues still :) After you fix those and the test, should be good to go!

zelioluca commented:
Hello there i do it now 😁

On Tue, 24 Oct 2023, 13:16 Niko Aarnio, ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi! The core functionality looks good to me and overall the style is
> almost according to the guideline, but I found a few small nitpicky style
> issues still :) After you fix those and the test, should be good to go!
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369911811>
> :
>
> > @@ -0,0 +1,48 @@
> +import numpy as np
> +import pandas
> +import pandas as pd
> +import sklearn.neural_network
> +from sklearn.inspection import permutation_importance
> +
> +from eis_toolkit.exceptions import InvalidDatasetException
> +
> +
> +def evaluate_feature_importance(
> +    clf: sklearn.neural_network or sklearn.linear_model,
>
> Could be named "classifier" so its more according to our naming
> conventions. If this accepts a sklearn classifier, should we give a union
> of the classifiers included in EIS Toolkit as the typing hint? Or perhaps
> sklearn BaseEstimator and then check if the given model is a classifier
> (using is_classifier method)? Also, already asking here, but did you do
> your MLP and CNN using sklearn or keras? I just realized maybe you used
> sklearn while I think the others have thought of using Keras...
>
> EDIT: just noticed this accepts also linear models, should this be
> reflected in the name and then just use sklearn BaseEstimator as hint?
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369912414>
> :
>
> > @@ -0,0 +1,48 @@
> +import numpy as np
> +import pandas
> +import pandas as pd
> +import sklearn.neural_network
> +from sklearn.inspection import permutation_importance
> +
> +from eis_toolkit.exceptions import InvalidDatasetException
> +
> +
> +def evaluate_feature_importance(
> +    clf: sklearn.neural_network or sklearn.linear_model,
> +    x_test: np.ndarray,
> +    y_test: np.ndarray,
> +    feature_names: list[str],
>
> Is any sequence ok here? We have used Sequence[str] for many cases before
> to allow tuples etc.
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369913204>
> :
>
> > +import numpy as np
> +import pandas
> +import pandas as pd
> +import sklearn.neural_network
> +from sklearn.inspection import permutation_importance
> +
> +from eis_toolkit.exceptions import InvalidDatasetException
> +
> +
> +def evaluate_feature_importance(
> +    clf: sklearn.neural_network or sklearn.linear_model,
> +    x_test: np.ndarray,
> +    y_test: np.ndarray,
> +    feature_names: list[str],
> +    number_of_repetition: int = 50,
> +    random_state: int = 0,
>
> Should the default be 0 or None? I thought None would be good default as
> it is so in Sklearn, but just asking
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369916058>
> :
>
> > @@ -0,0 +1,48 @@
> +import numpy as np
> +import pandas
>
> We have used import pandas as pd so far (and I noticed there seems to be
> duplicate import). Sorry for nitcpicky comment, but if its ok you should
> use the same import alias as elsewhere
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369920310>
> :
>
> > +from eis_toolkit.exceptions import InvalidDatasetException
> +
> +
> +def evaluate_feature_importance(
> +    clf: sklearn.neural_network or sklearn.linear_model,
> +    x_test: np.ndarray,
> +    y_test: np.ndarray,
> +    feature_names: list[str],
> +    number_of_repetition: int = 50,
> +    random_state: int = 0,
> +) -> (pandas.DataFrame, dict):
> +    """
> +    Evaluate the feature importance of a sklearn classifier or linear model.
> +
> +    Parameters:
> +        clf (Any sklearn nn model or lm model): Trained classifier.
>
> Regarding the docstring, no need to include the type in paranthesis (just
> a decision we made).
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369922147>
> :
>
> > +        random_state (int): random state for repeatability of results (Default 0).
> +    Return:
>
> ⬇️ Suggested change
>
> -        random_state (int): random state for repeatability of results (Default 0).
> -    Return:
> +        random_state (int): random state for repeatability of results (Default 0).
> +
> +    Return:
>
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369922448>
> :
>
> > +    random_state: int = 0,
> +) -> (pandas.DataFrame, dict):
> +    """
> +    Evaluate the feature importance of a sklearn classifier or linear model.
> +
> +    Parameters:
> +        clf (Any sklearn nn model or lm model): Trained classifier.
> +        x_test (np.ndarray): Testing feature data (X data need to be normalized / standardized).
> +        y_test (np.ndarray): Testing target data.
> +        feature_names (list): Names of the feature columns.
> +        number_of_repetition (int): Number of iteration used when calculate feature importance (default 50).
> +        random_state (int): random state for repeatability of results (Default 0).
> +    Return:
> +        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
> +        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
> +    Raise:
>
> ⬇️ Suggested change
>
> -    Raise:
> +
> +    Raises:
>
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369923339>
> :
>
> > +    Raise:
> +        InvalidDatasetException: When the dataset is None.
> +    """
> +
> +    if x_test is None or y_test is None:
> +        raise InvalidDatasetException
> +
> +    result = permutation_importance(
> +        clf, x_test, y_test.ravel(), n_repeats=number_of_repetition, random_state=random_state
> +    )
> +
> +    feature_importance = pd.DataFrame({"Feature": feature_names, "Importance": result.importances_mean})
> +
> +    feature_importance["Importance"] = feature_importance["Importance"] * 100
> +    feature_importance = feature_importance.sort_values(by="Importance", ascending=False)
> +    # feature_importance['Importance'] = feature_importance['Importance'].apply(lambda x: '{:.6f}%'.format(x))
>
> Could remove this if its not needed for final implementation
> ------------------------------
>
> In eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369923905>
> :
>
> > +        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
> +        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
>
> The names of the returned objects don't need to be stated, you can check
> some other docstrings for examples.
> ------------------------------
>
> On tests/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369925481>
> :
>
> The test file should use pytest and have some basic unit type tests.
> These are sometimes quite stupid to do for models, but you could load some
> test data or create some small test array and then see that the results are
> as you would expect. Again I suggest to take a look at some other test
> files.
>
> Some features are nice to demonstrate as a notebook, but if this does not
> deserve its own one in the toolkit (up to you), we can combine some
> modeling related stuff in one notebook to verify they work and show people
> how to use them at the same time.
> ------------------------------
>
> On .idea/eis_toolkit.iml
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369927499>
> :
>
> Not sure what is the .idea folder, but you should probably remove it from
> the PR
> ------------------------------
>
> On eis_toolkit/feature_importance/feature_importance.py
> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369928409>
> :
>
> Feature importance is marked in the excel at least to belong to
> Exploratory analyses, so you should move it there. Or if you think some
> other is more suitable thats ok too, but it shouldnt have its own folder
> directly under eis_toolkit folder
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694457705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF3KUFP6SAKIMSTHKR5HQP3YA6IQFAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUGQ2TONZQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

zelioluca commented:
Hi made all except pytest! Let me think little bit whatvi can do here 😁
anything else is complete

On Tue, 24 Oct 2023, 13:18 luca zelioli, ***@***.***> wrote:

> Hello there i do it now 😁
>
> On Tue, 24 Oct 2023, 13:16 Niko Aarnio, ***@***.***> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Hi! The core functionality looks good to me and overall the style is
>> almost according to the guideline, but I found a few small nitpicky style
>> issues still :) After you fix those and the test, should be good to go!
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369911811>
>> :
>>
>> > @@ -0,0 +1,48 @@
>> +import numpy as np
>> +import pandas
>> +import pandas as pd
>> +import sklearn.neural_network
>> +from sklearn.inspection import permutation_importance
>> +
>> +from eis_toolkit.exceptions import InvalidDatasetException
>> +
>> +
>> +def evaluate_feature_importance(
>> +    clf: sklearn.neural_network or sklearn.linear_model,
>>
>> Could be named "classifier" so its more according to our naming
>> conventions. If this accepts a sklearn classifier, should we give a union
>> of the classifiers included in EIS Toolkit as the typing hint? Or perhaps
>> sklearn BaseEstimator and then check if the given model is a classifier
>> (using is_classifier method)? Also, already asking here, but did you do
>> your MLP and CNN using sklearn or keras? I just realized maybe you used
>> sklearn while I think the others have thought of using Keras...
>>
>> EDIT: just noticed this accepts also linear models, should this be
>> reflected in the name and then just use sklearn BaseEstimator as hint?
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369912414>
>> :
>>
>> > @@ -0,0 +1,48 @@
>> +import numpy as np
>> +import pandas
>> +import pandas as pd
>> +import sklearn.neural_network
>> +from sklearn.inspection import permutation_importance
>> +
>> +from eis_toolkit.exceptions import InvalidDatasetException
>> +
>> +
>> +def evaluate_feature_importance(
>> +    clf: sklearn.neural_network or sklearn.linear_model,
>> +    x_test: np.ndarray,
>> +    y_test: np.ndarray,
>> +    feature_names: list[str],
>>
>> Is any sequence ok here? We have used Sequence[str] for many cases
>> before to allow tuples etc.
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369913204>
>> :
>>
>> > +import numpy as np
>> +import pandas
>> +import pandas as pd
>> +import sklearn.neural_network
>> +from sklearn.inspection import permutation_importance
>> +
>> +from eis_toolkit.exceptions import InvalidDatasetException
>> +
>> +
>> +def evaluate_feature_importance(
>> +    clf: sklearn.neural_network or sklearn.linear_model,
>> +    x_test: np.ndarray,
>> +    y_test: np.ndarray,
>> +    feature_names: list[str],
>> +    number_of_repetition: int = 50,
>> +    random_state: int = 0,
>>
>> Should the default be 0 or None? I thought None would be good default as
>> it is so in Sklearn, but just asking
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369916058>
>> :
>>
>> > @@ -0,0 +1,48 @@
>> +import numpy as np
>> +import pandas
>>
>> We have used import pandas as pd so far (and I noticed there seems to be
>> duplicate import). Sorry for nitcpicky comment, but if its ok you should
>> use the same import alias as elsewhere
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369920310>
>> :
>>
>> > +from eis_toolkit.exceptions import InvalidDatasetException
>> +
>> +
>> +def evaluate_feature_importance(
>> +    clf: sklearn.neural_network or sklearn.linear_model,
>> +    x_test: np.ndarray,
>> +    y_test: np.ndarray,
>> +    feature_names: list[str],
>> +    number_of_repetition: int = 50,
>> +    random_state: int = 0,
>> +) -> (pandas.DataFrame, dict):
>> +    """
>> +    Evaluate the feature importance of a sklearn classifier or linear model.
>> +
>> +    Parameters:
>> +        clf (Any sklearn nn model or lm model): Trained classifier.
>>
>> Regarding the docstring, no need to include the type in paranthesis (just
>> a decision we made).
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369922147>
>> :
>>
>> > +        random_state (int): random state for repeatability of results (Default 0).
>> +    Return:
>>
>> ⬇️ Suggested change
>>
>> -        random_state (int): random state for repeatability of results (Default 0).
>> -    Return:
>> +        random_state (int): random state for repeatability of results (Default 0).
>> +
>> +    Return:
>>
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369922448>
>> :
>>
>> > +    random_state: int = 0,
>> +) -> (pandas.DataFrame, dict):
>> +    """
>> +    Evaluate the feature importance of a sklearn classifier or linear model.
>> +
>> +    Parameters:
>> +        clf (Any sklearn nn model or lm model): Trained classifier.
>> +        x_test (np.ndarray): Testing feature data (X data need to be normalized / standardized).
>> +        y_test (np.ndarray): Testing target data.
>> +        feature_names (list): Names of the feature columns.
>> +        number_of_repetition (int): Number of iteration used when calculate feature importance (default 50).
>> +        random_state (int): random state for repeatability of results (Default 0).
>> +    Return:
>> +        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
>> +        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
>> +    Raise:
>>
>> ⬇️ Suggested change
>>
>> -    Raise:
>> +
>> +    Raises:
>>
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369923339>
>> :
>>
>> > +    Raise:
>> +        InvalidDatasetException: When the dataset is None.
>> +    """
>> +
>> +    if x_test is None or y_test is None:
>> +        raise InvalidDatasetException
>> +
>> +    result = permutation_importance(
>> +        clf, x_test, y_test.ravel(), n_repeats=number_of_repetition, random_state=random_state
>> +    )
>> +
>> +    feature_importance = pd.DataFrame({"Feature": feature_names, "Importance": result.importances_mean})
>> +
>> +    feature_importance["Importance"] = feature_importance["Importance"] * 100
>> +    feature_importance = feature_importance.sort_values(by="Importance", ascending=False)
>> +    # feature_importance['Importance'] = feature_importance['Importance'].apply(lambda x: '{:.6f}%'.format(x))
>>
>> Could remove this if its not needed for final implementation
>> ------------------------------
>>
>> In eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369923905>
>> :
>>
>> > +        feature_importance (pd.Dataframe): A dataframe composed by features name and Importance value
>> +        result (dict[object]): The resulted object with importance mean, importance std, and overall importance
>>
>> The names of the returned objects don't need to be stated, you can check
>> some other docstrings for examples.
>> ------------------------------
>>
>> On tests/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369925481>
>> :
>>
>> The test file should use pytest and have some basic unit type tests.
>> These are sometimes quite stupid to do for models, but you could load some
>> test data or create some small test array and then see that the results are
>> as you would expect. Again I suggest to take a look at some other test
>> files.
>>
>> Some features are nice to demonstrate as a notebook, but if this does not
>> deserve its own one in the toolkit (up to you), we can combine some
>> modeling related stuff in one notebook to verify they work and show people
>> how to use them at the same time.
>> ------------------------------
>>
>> On .idea/eis_toolkit.iml
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369927499>
>> :
>>
>> Not sure what is the .idea folder, but you should probably remove it
>> from the PR
>> ------------------------------
>>
>> On eis_toolkit/feature_importance/feature_importance.py
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1369928409>
>> :
>>
>> Feature importance is marked in the excel at least to belong to
>> Exploratory analyses, so you should move it there. Or if you think some
>> other is more suitable thats ok too, but it shouldnt have its own folder
>> directly under eis_toolkit folder
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694457705>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AF3KUFP6SAKIMSTHKR5HQP3YA6IQFAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUGQ2TONZQGU>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

nmaarnio commented:
Forgot one thing from the review: please add the `@beartype` decorator for input checking!

nmaarnio commented on eis_toolkit/feature_importance/__init__.py:
Also please delete this folder

nmaarnio commented on tests/exploratory_analyses/feature_importance_test.py:
```diff
@@ -0,0 +1,59 @@
+    "EM_Inph",
+]
+
+X = pd.read_csv("../data/local/data/fake_smote_data.csv").to_numpy()
```
If you put this dataset to somewhere in the data/remote, then the test will work for everyone. Now the dataset stays on your computer.

zelioluca replied:
I puty it in remote folder

nmaarnio reviewed (commented):
Only a few files/folder should be removed from the PR and test data put to remote, then we can merge!

zelioluca commented:
Give me file

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Niko Aarnio ***@***.***>
Sent: Tuesday, October 24, 2023 3:54:51 PM
To: GispoCoding/eis_toolkit ***@***.***>
Cc: Luca Zelioli ***@***.***>; Author ***@***.***>
Subject: Re: [GispoCoding/eis_toolkit] I added the feature importance function (PR #211)


@nmaarnio commented on this pull request.

Only a few files/folder should be removed from the PR and test data put to remote, then we can merge!

________________________________

On .idea/eis_toolkit.iml<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370108657>:

Could you remove these files still? I saw you added the folder in gitignore but the folder and files still remain.

________________________________

On eis_toolkit/feature_importance/__init__.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370109020>:

Also please delete this folder

________________________________

In tests/exploratory_analyses/feature_importance_test.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370110069>:

> +    "HDTDR",
+    "Mag_Xdrv",
+    "mag_Ydrv",
+    "Mag_Zdrv",
+    "Pseu_Grv",
+    "Rd_U",
+    "Rd_TC",
+    "Rd_Th",
+    "Rd_K",
+    "EM_ratio",
+    "EM_Ap_rs",
+    "Em_Qd",
+    "EM_Inph",
+]
+
+X = pd.read_csv("../data/local/data/fake_smote_data.csv").to_numpy()


If you put this dataset to somewhere in the data/remote, then the test will work for everyone. Now the dataset stays on your computer.

—
Reply to this email directly, view it on GitHub<https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694761344>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AF3KUFIIFJWXQMV3OWP26O3YA63BXAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUG43DCMZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zelioluca commented:
I m doing the test now

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Niko Aarnio ***@***.***>
Sent: Tuesday, October 24, 2023 3:54:51 PM
To: GispoCoding/eis_toolkit ***@***.***>
Cc: Luca Zelioli ***@***.***>; Author ***@***.***>
Subject: Re: [GispoCoding/eis_toolkit] I added the feature importance function (PR #211)


@nmaarnio commented on this pull request.

Only a few files/folder should be removed from the PR and test data put to remote, then we can merge!

________________________________

On .idea/eis_toolkit.iml<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370108657>:

Could you remove these files still? I saw you added the folder in gitignore but the folder and files still remain.

________________________________

On eis_toolkit/feature_importance/__init__.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370109020>:

Also please delete this folder

________________________________

In tests/exploratory_analyses/feature_importance_test.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370110069>:

> +    "HDTDR",
+    "Mag_Xdrv",
+    "mag_Ydrv",
+    "Mag_Zdrv",
+    "Pseu_Grv",
+    "Rd_U",
+    "Rd_TC",
+    "Rd_Th",
+    "Rd_K",
+    "EM_ratio",
+    "EM_Ap_rs",
+    "Em_Qd",
+    "EM_Inph",
+]
+
+X = pd.read_csv("../data/local/data/fake_smote_data.csv").to_numpy()


If you put this dataset to somewhere in the data/remote, then the test will work for everyone. Now the dataset stays on your computer.

—
Reply to this email directly, view it on GitHub<https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694761344>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AF3KUFIIFJWXQMV3OWP26O3YA63BXAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUG43DCMZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zelioluca commented:
How i put the path of that folder like this test does not like

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Niko Aarnio ***@***.***>
Sent: Tuesday, October 24, 2023 3:54:51 PM
To: GispoCoding/eis_toolkit ***@***.***>
Cc: Luca Zelioli ***@***.***>; Author ***@***.***>
Subject: Re: [GispoCoding/eis_toolkit] I added the feature importance function (PR #211)


@nmaarnio commented on this pull request.

Only a few files/folder should be removed from the PR and test data put to remote, then we can merge!

________________________________

On .idea/eis_toolkit.iml<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370108657>:

Could you remove these files still? I saw you added the folder in gitignore but the folder and files still remain.

________________________________

On eis_toolkit/feature_importance/__init__.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370109020>:

Also please delete this folder

________________________________

In tests/exploratory_analyses/feature_importance_test.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370110069>:

> +    "HDTDR",
+    "Mag_Xdrv",
+    "mag_Ydrv",
+    "Mag_Zdrv",
+    "Pseu_Grv",
+    "Rd_U",
+    "Rd_TC",
+    "Rd_Th",
+    "Rd_K",
+    "EM_ratio",
+    "EM_Ap_rs",
+    "Em_Qd",
+    "EM_Inph",
+]
+
+X = pd.read_csv("../data/local/data/fake_smote_data.csv").to_numpy()


If you put this dataset to somewhere in the data/remote, then the test will work for everyone. Now the dataset stays on your computer.

—
Reply to this email directly, view it on GitHub<https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694761344>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AF3KUFIIFJWXQMV3OWP26O3YA63BXAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUG43DCMZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zelioluca commented:
Yes i fixed it 😁

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Niko Aarnio ***@***.***>
Sent: Tuesday, October 24, 2023 3:54:51 PM
To: GispoCoding/eis_toolkit ***@***.***>
Cc: Luca Zelioli ***@***.***>; Author ***@***.***>
Subject: Re: [GispoCoding/eis_toolkit] I added the feature importance function (PR #211)


@nmaarnio commented on this pull request.

Only a few files/folder should be removed from the PR and test data put to remote, then we can merge!

________________________________

On .idea/eis_toolkit.iml<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370108657>:

Could you remove these files still? I saw you added the folder in gitignore but the folder and files still remain.

________________________________

On eis_toolkit/feature_importance/__init__.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370109020>:

Also please delete this folder

________________________________

In tests/exploratory_analyses/feature_importance_test.py<https://github.com/GispoCoding/eis_toolkit/pull/211#discussion_r1370110069>:

> +    "HDTDR",
+    "Mag_Xdrv",
+    "mag_Ydrv",
+    "Mag_Zdrv",
+    "Pseu_Grv",
+    "Rd_U",
+    "Rd_TC",
+    "Rd_Th",
+    "Rd_K",
+    "EM_ratio",
+    "EM_Ap_rs",
+    "Em_Qd",
+    "EM_Inph",
+]
+
+X = pd.read_csv("../data/local/data/fake_smote_data.csv").to_numpy()


If you put this dataset to somewhere in the data/remote, then the test will work for everyone. Now the dataset stays on your computer.

—
Reply to this email directly, view it on GitHub<https://github.com/GispoCoding/eis_toolkit/pull/211#pullrequestreview-1694761344>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AF3KUFIIFJWXQMV3OWP26O3YA63BXAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOJUG43DCMZUGQ>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

zelioluca commented:
tests passed finally

nmaarnio commented:
Looks good now, merging!

zelioluca commented:
Tomorrow i do cnn

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Niko Aarnio ***@***.***>
Sent: Tuesday, October 24, 2023 4:58:43 PM
To: GispoCoding/eis_toolkit ***@***.***>
Cc: Luca Zelioli ***@***.***>; Author ***@***.***>
Subject: Re: [GispoCoding/eis_toolkit] I added the feature importance function (PR #211)


Looks good now, merging!

—
Reply to this email directly, view it on GitHub<https://github.com/GispoCoding/eis_toolkit/pull/211#issuecomment-1777269253>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AF3KUFNKHCZIIBXG2MJMAPTYA7CRHAVCNFSM6AAAAAA6NLHFNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONZXGI3DSMRVGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
